### ISSUE TITLE
[Feature Request] Support for SSH Keys, and Dedicated Servers in general

### ISSUE THREAD

**Issue by BenCat07** [thommcgrath/Beacon#271]
As in the title, i think it would be sufficient to allow adjusting the SFTP option to firstly allow giving a root directory for ARK (having it be optional though), and secondly, but probably much more annoyingly, add support for adding an SSH key to make a connection to the server. That functionality is lacking, and makes it impossible to use any sort of connection options for a set to only allow connections made with a proper SSH key.

Example for me:
I have a server i have running myself where i installed ARK, which is in ~/ARK2 (don't ask what happened to just ~/ARK). Currently i have to manually either sftp the GUS.ini and Game.ini over, or straight up use `cat` and put it in my clipboard. Applying is the reverse, with deleting the file, or making a backup, then recreating the file with the new Backup. Would be nice to be able to manage it Automatically from Beacon, like with Hosts specifically for Ark.

I understand that this is probably very low priority, as most Beacon users will not use self-hosted server, but it would still be nice to have for the few people that do :)

In Short:
- Support for SSH key authentication instead of Password in sftp
- Ability to give a root directory to search in sftp, for the purpose of Beacon using the proper directory

**Comment by thommcgrath** [comment]
Private key authentication is definitely lacking, but I can't remember the roadblock I ran into, so I'll take another look. Beacon uses cURL, which will automatically use `~/.ssh/id_dsa` and `~/.ssh/id_dsa.pub`, but I'm guessing you're using RSA or ED2559 like any modern system should be.

As for the path, can you explain more about your setup? Beacon will already ask for your Game.ini path when you import the server, so I'm not understanding what it is you're asking for.

**Comment by BenCat07** [comment]
Ah i see, that is my bad then, no problem with that. The issue was that i wasn't sure how to handle connecting to my server in the first place, and did not see an option to pass path directly on connection. I'll remove that from the issue.

**Comment by thommcgrath** [comment]
For an RSA key, yes. But the docs say `id_dsa` for DSA key is the default. Which is strange because DSA isn’t a great choice for SSH.
